packages/x-components/src/x-modules/history-queries/store/__tests__/actions.spec.ts

```diff
@@ -269,4 +269,88 @@ describe('testing history queries module actions', () => {
       expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(true);
     });
   });
+
+  describe('updateHistoryQueriesWithSearchResponse', () => {
+    const results: Result[] = [
+      { modelName: 'Result', id: '1' },
+      { modelName: 'Result', id: '2' }
+    ];
+    const totalResults = results.length;
+    let gato: HistoryQuery, perro: HistoryQuery;
+
+    beforeEach(() => {
+      [gato, perro] = createHistoryQueries('gato', 'perro');
+      resetStateWith({ historyQueries: [gato, perro] });
+    });
+
+    it('updates a history query if its response is a success', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1
+        },
+        status: 'success',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([{ ...gato, totalResults }, perro]);
+    });
+
+    it('does not update a history query if its response is an error', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'gato',
+          page: 1
+        },
+        status: 'error',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([gato, perro]);
+    });
+
+    it('does not update a history query if it is not stored', async () => {
+      await store.dispatch('updateHistoryQueriesWithSearchResponse', {
+        request: {
+          query: 'pato',
+          page: 1
+        },
+        status: 'success',
+        results,
+        totalResults
+      });
+      expectHistoryQueriesToEqual([gato, perro]);
+    });
+
+    it('updates a history query if it was made on a previous session', async () => {
+      gato.totalResults = 50;
+      gato.timestamp -= store.state.sessionTimeStampInMs;
```

Comment:
I would set it to `store.state.sessionTimeStampinMs - 1` because in the tests `sessionTimeStampinMs` seems to be 0.

```suggestion
      gato.timestamp = store.state.sessionTimeStampInMs - 1;
```

The test is passing right now, but I think it should be failing, I will let you another comment in the action.

Reply:
Ok, since the `sessionTimeStampInMs` is not considered to be in the current session, what do you think about setting the query timestamp to the same value?
So it is clear that this behavior is intentional.

```suggestion
      gato.timestamp = store.state.sessionTimeStampInMs;
```